WordPress/Classes/ViewRelated/System/Coordinators/MySitesCoordinator.swift

```diff
@@ -219,4 +225,19 @@ class MySitesCoordinator: NSObject {
 
         navigationController.pushViewController(listViewController, animated: false)
     }
+
+    // MARK: Notifications Handling
+
+    private func addSignInObserver() {
+        let notificationName = NSNotification.Name(WordPressAuthenticator.WPSigninDidFinishNotification)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(signinDidFinish),
+                                               name: notificationName,
+                                               object: nil)
+    }
+
+    @objc func signinDidFinish() {
+        mySiteViewController = makeMySiteViewController()
```

Comment:
I tested various scenarios signing up, adding self-hosted website, simple sign-ins and it worked well ✅